integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/sql/FlowConrolFunctionIT.java

```diff
@@ -0,0 +1,121 @@
+import org.json.JSONObject;
+
+import org.junit.Assume;
+import org.junit.Test;
+
+
+/**
+ * Created by allwefantasy on 8/25/16.
+ */
+public class FlowConrolFunctionIT extends SQLIntegTestCase {
+
+  @Override
+  public void init() throws Exception {
+    super.init();
+    TestUtils.enableNewQueryEngine(client());
+    loadIndex(Index.BANK);
+    loadIndex(Index.ACCOUNT);
+    loadIndex(Index.ONLINE);
+    loadIndex(Index.DATE);
+  }
+
+  @Test
+  public void ifnullShouldPassJDBC() throws IOException {
+    JSONObject response = executeJdbcRequest(
+        "SELECT IFNULL(lastname, 'unknown') AS name FROM " + TEST_INDEX_ACCOUNT
+            + " GROUP BY name");
+    assertEquals("IFNULL(lastname, \'unknown\')", response.query("/schema/0/name"));
+    assertEquals("name", response.query("/schema/0/alias"));
+    assertEquals("keyword", response.query("/schema/0/type"));
+  }
+
+  @Test
+  public void ifnullWithNotNullInputTest() throws IOException {
+    assertThat(
+        executeQuery("SELECT IFNULL('sample', 'IsNull') AS ifnull FROM " + TEST_INDEX_ACCOUNT),
+        hitAny(kvString("/fields/ifnull/0", equalTo("sample")))
+    );
+  }
+
+  @Test
+  public void ifnullWithNullInputTest() throws IOException {
+    assertThat(
+        executeQuery("SELECT IFNULL(null, 10) AS ifnull FROM " + TEST_INDEX_ACCOUNT),
+        hitAny(kvInt("/fields/ifnull/0", equalTo(10)))
+    );
+    assertThat(
+        executeQuery("SELECT IFNULL('', 10) AS ifnull FROM " + TEST_INDEX_ACCOUNT),
+        hitAny(kvString("/fields/ifnull/0", equalTo("")))
+    );
+  }
+
+  @Test
+  public void nullifShouldPassJDBC() throws IOException {
+    Assume.assumeTrue(isNewQueryEngineEabled());
+    JSONObject response = executeJdbcRequest(
+            "SELECT NULLIF(lastname, 'unknown') AS name FROM " + TEST_INDEX_ACCOUNT);
+    assertEquals("NULLIF(lastname, \'unknown\')", response.query("/schema/0/name"));
+    assertEquals("name", response.query("/schema/0/alias"));
+    assertEquals("keyword", response.query("/schema/0/type"));
+  }
+
+  @Test
+  public void isnullShouldPassJDBC() throws IOException {
+    Assume.assumeTrue(isNewQueryEngineEabled());
+    JSONObject response = executeJdbcRequest(
+            "SELECT ISNULL(lastname) AS name FROM " + TEST_INDEX_ACCOUNT);
+    assertEquals("ISNULL(lastname)", response.query("/schema/0/name"));
+    assertEquals("name", response.query("/schema/0/alias"));
+    assertEquals("boolean", response.query("/schema/0/type"));
+  }
+
+  private SearchHits query(String query) throws IOException {
+    final String rsp = executeQueryWithStringOutput(query);
+
+    final XContentParser parser = XContentFactory.xContent(XContentType.JSON).createParser(
+        NamedXContentRegistry.EMPTY,
+        LoggingDeprecationHandler.INSTANCE,
+        rsp);
+    return SearchResponse.fromXContent(parser).getHits();
+  }
+
+  private JSONObject executeJdbcRequest(String query) {
+    return new JSONObject(executeQuery(query, "jdbc"));
+  }
```

Comment:
Could you use the `executeQuery` method inherited from the parent class `SQLIntegTestCase` to execute SQL queries, it will enforce to JDBC format by default. For the assertion, you can check out the `verifySchema`, `verifyDataRows` etc. methods in `MatcherUtils`. For example:
```
JSONObject result = executeQuery(SELECT ISNULL(lastname) AS name ...);
verifySchema(result, schema("ISNULL(lastname)"), "name" , "boolean");
verifyDataRows(...);
```
You can also checkout the IT cases in `DataTimeFunctionIT.java` for reference.

Reply:
Got it, thanks